bibliopixel/drivers/spi_driver_base.py

```diff
@@ -2,76 +2,111 @@
         super().__init__(num, c_order=c_order, gamma=gamma)
 
+        self.open = open
+        self.exists = exists
+        print(self.exists)
+
         self.dev = dev
-        self.use_py_spi = use_py_spi
+        self.interface = interface
+        self.numSPI = -1  # ID of SPI device
+        self.numCS = -1  # ID of Chip Select
         self._spiSpeed = SPISpeed
 
-        # File based SPI requires a bytearray so we have to overwrite _buf
-        if self.use_py_spi:
-            a, b = -1, -1
-            d = self.dev.replace("/dev/spidev", "")
-            s = d.split('.')
-            if len(s) == 2:
-                a = int(s[0])
-                b = int(s[1])
+        self._bootstrapSPIDev()
 
-            if a < 0 or b < 0:
-                error = "When using py-spidev, the given device must be in the format /dev/spidev*.*"
-                log.error(error)
-                raise ValueError(error)
+        # if SpiInterface is not File we need a specific format for the python api
+        if self.interface != SPIInterface.FILE and (self.numSPI < 0 or self.numCS < 0):
+            error = "When using py-spidev, the given device must be in the format /dev/spidev*.*"
+            log.error(error)
+            raise ValueError(error)
 
-            self._bootstrapSPIDev()
+        # File based SPI requires a bytearray so we have to overwrite _buf
+        if self.interface == SPIInterface.PY_SPI:
             import spidev
             self.spi = spidev.SpiDev()
-            self.spi.open(a, b)
+            self.spi.open(self.numSPI, self.numCS)
             self.spi.max_speed_hz = int(self._spiSpeed * 1000000.0)
             log.info('py-spidev speed @ %.1f MHz',
                      (float(self.spi.max_speed_hz) / 1000000.0))
-        else:
+        elif self.interface == SPIInterface.PI_GPIO:
+            # @TODO support chip select http://abyz.co.uk/rpi/pigpio/python.html#spi_open
+            AUX_SPI = 8
+            flags = (1 << AUX_SPI)
+            import pigpio
+            self.pi = pigpio.pi()
+            self.spi = self.pi.spi_open(self.numSPI, int(self._spiSpeed * 1000000.0), flags)
+            log.info('py-spidev speed @ %.1f MHz', (float(self.spi.max_speed_hz) / 1000000.0))
+        elif self.interface == SPIInterface.FILE:
             self._buf = bytearray(self._buf)
-            self.spi = open(self.dev, "wb")
+            self.spi = self.open(self.dev, "wb")
+        else:
+            raise ValueError('invalid interface type: "%s" not found ' % self.interface)
 
     def _bootstrapSPIDev(self):
-        import os.path
-        import sys
-        if self.use_py_spi:
+        if self.interface != SPIInterface.FILE:
+            self.numSPI, self.numCS = -1, -1
+            d = self.dev.replace("/dev/spidev", "")
+            s = d.split('.')
+            if len(s) == 2:
+                self.numSPI = int(s[0])
+                self.numCS = int(s[1])
```

Comment:
If s doesn't length 2, you fail, silently!

I'd go with `self.numSPI, self.numCS = (int(i) for i in s)` which throws an exception